### PR TITLE
Check if assigned string  is valid in name setter method of Register class

### PR DESCRIPTION
**Pull request by avartak**
### Summary
This PR addresses the issue raised in [#5461](https://github.com/Qiskit/qiskit-terra/issues/5461)

The @name.setter method of Register class currently does not check if the assigned string
is valid as per the OPENQASM requirements. This check is, however, performed in the constructor.

For consistency, the check has now been added to the name setter method

### Details and comments

In the @name.setter method, if the string being assigned to the register name is not OPENQASM compliant i.e. it does not conform to the format [a-z][a-zA-Z0-9_]* then a CircuitError is raised.

### CODE REVIEW

**Review by kdk — COMMENTED** [review]
Thanks @avartak , the fix looks good. Looks like sphinx is unhappy with the `jupyter-execute` in the release note ( https://dev.azure.com/qiskit-ci/qiskit-terra/_build/results?buildId=22028&view=logs&j=23e88340-9b01-571b-4128-239215012d30&t=fd5b7ae3-9f70-5b67-b24e-b010b8098eb2&l=121 ). Maybe you can convert it to a code block?

**Comment by avartak** [comment]
Thanks for the suggestion @kdk . I changed the `jupyter-execute` to a `code-block` and that seems to have fixed the problem.

File: qiskit/circuit/register.py [hunk 1]
```diff
@@ -84,6 +84,8 @@ def name(self):
     @name.setter
     def name(self, value):
         """Set the register name."""
+        if self.name_format.match(value) is None:
+            raise CircuitError("%s is an invalid OPENQASM register name." % value)
```
Comment (Cryoris) [inline qiskit/circuit/register.py:88]: Could we add a reference where users could look up valid names?

Comment (avartak) [reply qiskit/circuit/register.py]: Thanks for the suggestion @Cryoris . I have added the link to the Open QASM arXiv document to the error message. I have also mentioned it in the release note.

**Review by kdk — APPROVED** [review]
Thanks for the updates!

**Pull request merged**